src/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -1026,25 +1008,24 @@ internal sealed class QueueUserWorkItemCallbackDefaultContext : QueueUserWorkIte
         private WaitCallback _callback;
         private readonly object _state;
 
-        internal static readonly Action<QueueUserWorkItemCallbackDefaultContext> s_executionContextShim = quwi =>
+        internal QueueUserWorkItemCallbackDefaultContext(WaitCallback callback, object state)
         {
-            WaitCallback callback = quwi._callback;
             Debug.Assert(callback != null);
-            quwi._callback = null;
 
-            callback(quwi._state);
-        };
-
-        internal QueueUserWorkItemCallbackDefaultContext(WaitCallback callback, object state)
-        {
             _callback = callback;
             _state = state;
         }
 
         public override void Execute()
         {
             base.Execute();
-            ExecutionContext.RunForThreadPoolUserWorkItemWithDefaultContext(s_executionContextShim, this);
+
+            WaitCallback callback = _callback;
```

Comment:
Since this expects to run in the default context it would be good to assert that the current contexts are default (the `CheckThreadPoolAndContextsAreDefault` that was there before)

Reply:
Added (x3)